### PR TITLE
add by_source, by_lang, and by_length

### PR DESCRIPTION
**Pull request by jdorweiler**
I added a few more hash look up functions for fatheads.

### CODE REVIEW

File: lib/DDG/Meta/Data.pm [hunk 1]
```diff
@@ -181,6 +185,15 @@ sub get_js {
 # return a hash of IA objects by id
 sub by_id { $ia_metadata{id} }
 
+# return a hash of IA metadata for fatheads
+sub by_source { $ia_metadata{source} }
```
Comment (zachthompson) [inline lib/DDG/Meta/Data.pm]: Let's make these "by_fathead_*" so people don't see them being used, assume they work for all IAs, and wonder why they only get fathead data.

Comment (jdorweiler) [inline lib/DDG/Meta/Data.pm]: Done!

File: lib/DDG/Meta/Data.pm [hunk 2]
```diff
@@ -92,15 +92,19 @@ unless(%ia_metadata){
             $ia_metadata{id}{$id} = $module_data;
             #add new ia to ia_metadata{module}. Multiple ias per module possible
             push @{$ia_metadata{module}{$perl_module}}, $module_data;
-            # by language for multilang wiki
+
             if($is_fathead){
+                my $source = $module_data->{src_id};
                 # by source number for fatheads
-                $ia_metadata{source}{$module_data->{src_id}} = $module_data;
+                $ia_metadata{source}{$source} = $module_data;
```
Comment (zachthompson) [inline lib/DDG/Meta/Data.pm]: We might want to make these keys "fathead_source", "fathead_lang", etc.  Not a show stopper though.

Comment (jdorweiler) [inline lib/DDG/Meta/Data.pm]: that changes all the get_ia and get_js calls.

Comment (jdorweiler) [inline lib/DDG/Meta/Data.pm]: updated them

**Pull request merged**